packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -258,13 +258,29 @@ export default new Bundler({
       for (let asset of assets) {
         bundleGraph.addAssetGraphToBundle(asset, sharedBundle);
         for (let bundle of sourceBundles) {
-          bundleGraph.removeAssetGraphFromBundle(asset, bundle);
+          let bundleGroups = bundleGraph.getBundleGroupsContainingBundle(
+            bundle,
+          );
+          if (
```

Comment:
👍 May be worth extracting this as common logic with the parallel request check above